examples/TestEm3/electrons.cu

```diff
@@ -28,6 +28,9 @@
 #include <G4HepEmElectronEnergyLossFluctuation.icc>
 #endif
 
+__device__ char g_nextInteractionForEl[8'000'000];
+__device__ char g_nextInteractionForPos[8'000'000];
+
```

Comment:
As you write in the summary, this isn't very nice. What is the plan here, move the `winnerProcessIndex` into the `Track` structure?

Another approach (that I think we had in the past? not sure) is having queues for the discrete processes. That would even save us from determining which `Track` need to run in which of the (split) kernels.